12.0 project parent

gfcapalbo opened the pull request:
project parent v12

YogeshMahera-SerpentCS committed:
[ADD] - Added Project Parent module (#326)

YogeshMahera-SerpentCS committed:
[IMP/ADD]Added Child Project Smart button in parent project module

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update project_parent.pot

oca-travis committed:
[UPD] Update project_parent.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

Commit:
[MIG] Migrate project_parent from 10.0 to 12.0

gfcapalbo committed:
[UPD] [no task] parent_project v12

Saran440 commented:
@gfcapalbo please fix travis. In v.12 `# -*- coding: utf-8 -*- ` is not necessary

Functional test.
Name in Child Project should be `Child Projcet` instead of `Unnamed`

![Selection_010](https://user-images.githubusercontent.com/20896369/74526826-cf2f8080-4f56-11ea-8c33-b7af8a2e1d90.png)

gfcapalbo committed:
[FIX] naming
[FIX] utf remove

gfcapalbo commented:
@Saran440  thank you for testing
removed utf, added a name to the action "children of {parent}"
![screenshot_parent1](https://user-images.githubusercontent.com/8994879/76318625-4817c300-62de-11ea-93f3-307fcad6cee3.png)

janverb commented on project_parent/post_install.py:
```diff
@@ -0,0 +1,50 @@
+        'SELECT count(attname) FROM pg_attribute '
+        'WHERE attrelid = '
+        '( SELECT oid FROM pg_class WHERE relname = %s ) '
+        'AND attname = %s',
```
Nit: it would be nicer for this query and the one below to also use multi-line string literals.

janverb commented on project_parent/views/project_parent.xml:
```diff
@@ -0,0 +1,67 @@
+           </xpath>
+         </field>
+        </record>
+    -->
```
Should this be commented out?

janverb commented on project_parent/views/project_parent.xml:
```diff
@@ -0,0 +1,67 @@
+                <field name="parent_id"/>
+            </field>
+           <xpath expr="//filter[contains(@domain, user_id)]" position="after">
+              <filter string="Parent" domain="[]" context="{'group_by':'parent_id'}"/>
```
I'm guessing this is still the Odoo 10 version. I can get it working like this:
```suggestion
           <xpath expr="//filter[@name='Partner']" position="after">
              <filter string="Parent Project" name="Parent" context="{'group_by':'parent_id'}"/>
```

janverb commented on project_parent/models/project.py:
```diff
@@ -0,0 +1,29 @@
+    _parent_store = True
+
+    parent_id = fields.Many2one(
+       comodel_name='project.project', string='Parent Project'
```
`flake8` wants this to be indented one space more.

janverb commented on project_parent/models/project.py:
```diff
@@ -0,0 +1,29 @@
+       comodel_name='project.project', string='Parent Project'
+    )
+    child_ids = fields.One2many(comodel_name='project.project',
+                                inverse_name='parent_id')
```
These conflict with [fields in `project_wbs`](https://github.com/OCA/project/blob/1247db6a383ceb310403d787e9df00bb56dfe6af/project_wbs/models/project_project.py#L164), which point to `account.analytic.account`. So one of the modules has to change, I think. Changing this one would be less disruptive.

gfcapalbo replied:
changing it to parent_project_id

janverb commented on project_parent/views/project_parent.xml:
```diff
@@ -0,0 +1,67 @@
+            <field name="rating_status" position="after">
+                <field name="parent_id"/>
+            </field>
+            <xpath expr="//div[contains(@class, 'o_project_kanban_main')]/div[last()]" position="after">
```
Odoo complains that it's better to write it this way:
```suggestion
            <xpath expr="//div[hasclass('o_project_kanban_main')]/div[last()]" position="after">
```

janverb reviewed (changes requested):
This works well functionally. I had a few technical comments.

gfcapalbo committed:
[FIX] review fixes

gfcapalbo commented:
@janverb  thanks for the great review and the catch of the parent name conflict. implemented all your suggestions.

gfcapalbo committed:
[FIX]

tfossoul commented on project_parent/models/project.py:
```diff
@@ -0,0 +1,30 @@
+    )
+    child_ids = fields.One2many(comodel_name='project.project',
+                                inverse_name='project_parent_id')
+    parent_path = fields.Char(index=True)
```
What is it ?

Saran440 commented on project_parent/__manifest__.py:
```diff
@@ -0,0 +1,18 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+{
+    'name': 'Project Parent',
+    'version': '12.0.3.0.0',
```
```suggestion
    'version': '12.0.1.0.0',
```

Saran440 commented on project_parent/__manifest__.py:
```diff
@@ -0,0 +1,18 @@
+{
+    'name': 'Project Parent',
+    'version': '12.0.3.0.0',
+    'license': 'LGPL-3',
```
```suggestion
    'license': 'LGPL-3',
```
This module is LGPL or AGPL ?
I see license AGPL in file python but manifest is LGPL
`# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).`

Saran440 reviewed (commented):
Code Reviewed.

gfcapalbo committed:
[UPD] consistency in licensing and module version change

gfcapalbo commented:
Added consistency in licensing ,  set  version to 12.0.1.0.0, rebased.
